### PR TITLE
Update dependencies to TypeScript 3.7 and simplify some code

### PR DESCRIPTION
**Pull request by mrseanryan**
- updates TypeScript to 3.7
- simplifies some code, using the new optional chaining operator `?.`
- simplifies code by defaulting baseUrl to be '.' (with tests added)

### CODE REVIEW

**Comment by mrseanryan** [comment]
Hmm I cannot seem to avoid reducing the code coverage.

I think the *covered* code has shrunk with the use of `?.` etc and so the covered % is down a tiny bit.

File: src/parser/import.ts [hunk 1]
```diff
@@ -27,19 +27,26 @@ export const extractImport = (decl: ts.ImportDeclaration): FromWhat => {
...
+  if (!namedBindings) {
+    return {
+      from,
+      what: importDefault,
+    };
+  }
+
+  const isStar = !!(namedBindings as ts.NamespaceImport).name;
+  const importStar = isStar ? STAR : [];
+  const importNames = isStar
```
Comment (pzavolinsky) [inline src/parser/import.ts]: We could merge these two ternary operators, as they are mutually exclusive. Something like

```ts
const importNames =
  isStart
    ? STAR
    : (namedBindings as ts.NamedImports).elements.map(
        e => (e.propertyName || e.name).text,
      );
```

**Comment by pzavolinsky** [comment]
> But overall I think this is an improvement to the code.

Agreed, the code looks much better.

I only left you a small comment up there (something that was there before, but now it seems obvious we can improve).

Regarding the coverage, a drop of a 0.3% is not a big deal but just in case you could check the detailed coverage info to see what changed.

File: src/parser/import.ts (hunk 1 above)
Comment (mrseanryan) [reply src/parser/import.ts]: done (thanks for the quick review!)

**Comment by mrseanryan** [comment]
@pzavolinsky  thanks for reviewing.  

I checked the code coverage again - it is same old lines that look correct, and right now can't think of how to cover them.

I agree, the drop of .3% is not a big deal :)

Sean

**Comment by mrseanryan** [comment]
note: have configured coveralls to allow a decrease of less than 1%

(as failing on tiny increases is not useful)

**Pull request merged**